terraform/loadgen/00_loadgen.tf

```diff
@@ -0,0 +1,150 @@
+# back to that in a minute.
+#
+# One thing to note here is the name of the resource ("gke") is only used
+# internally, for instance when you're referencing the resource (eg
+# `google_container_cluster.gke.id`). The actual created resource won't know
+# about it, and in fact you can specify the name for that in the resource
+# itself.
+#
+# Finally, there are many, many other options available. The resource below
+# replicates what the Hipster Shop README creates. If you want to see what else
+# is possible, check out the docs: https://www.terraform.io/docs/providers/google/r/container_cluster.html
+resource "google_container_cluster" "gke_loadgen" {
+  # Here's how you specify the name of the cluster
+  name = "cloud-ops-sandbox-loadgenerator"
+
+  project = var.project_id
+
+  # Set the zone by grabbing the result of the random_shuffle above. It
+  # returns a list so we have to pull the first element off. If you're looking
+  # at this and thinking "huh terraform syntax looks a clunky" you are NOT WRONG
+  location = element(random_shuffle.zone.result, 0)
+
+  # Using an embedded resource to define the node pool. Another
+  # option would be to create the node pool as a separate resource and link it
+  # to this cluster. There are tradeoffs to each approach.
+  #
+  # The embedded resource is convenient but if you change it you have to tear
+  # down the entire cluster and rebuild it. A separate resource could be
+  # modified independent of the cluster without the cluster needing to be torn
+  # down.
+  #
+  # For this particular case we're not going to be modifying the node pool once
+  # it's deployed, so it makes sense to accept the tradeoff for the convenience
+  # of having it inline.
+  #
+  # Many of the paramaters below are self-explanatory so I'll only call out
+  # interesting things.
+  node_pool {
+    node_config {
+      machine_type = "n1-standard-2"
+
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/cloud-platform"  
+      ]
+
+      labels = {
+        environment = "dev",
+        cluster = "cloud-ops-sandbox-loadgenerator"
+      }
+    }
+
+    initial_node_count = 3
+
+    autoscaling {
+      min_node_count = 1
+      max_node_count = 10
+    }
+
+    management {
+      auto_repair  = true
+      auto_upgrade = true
+    }
+  }
+
+  # Stores the zone of created loadgenerator gke cluster
+  provisioner "local-exec" {
+    command = "gcloud config set compute/zone ${element(random_shuffle.zone.result, 0)}"
+  }
+}
+
+
+# Set current project 
+resource "null_resource" "current_project" {
+  provisioner "local-exec" {
+    command = "gcloud config set project ${var.project_id}"
+  }
+}
+
+# Setting kubectl context to currently deployed loadgenerator GKE cluster
+resource "null_resource" "set_gke_context" {
+  provisioner "local-exec" {
+    command = "gcloud container clusters get-credentials cloud-ops-sandbox-loadgenerator --zone ${element(random_shuffle.zone.result, 0)} --project ${var.project_id}"
```

Comment:
use variable substitution that evaluates to cloud-ops-sandbox-loadgenerator (instead of using a literal constant)

Reply:
it's not variable though, we name it in the terraform script? could put it in a constant i guess, but we do the exact same for cloud-ops-sandbox above